frontend/src/scenes/data-warehouse/editor/draftsLogic.tsx

```diff
@@ -0,0 +1,158 @@
+import { actions, kea, listeners, path, reducers, selectors } from 'kea'
+import api, { PaginatedResponse } from 'lib/api'
+
+import { DataWarehouseSavedQueryDraft } from '~/types'
+
+import type { draftsLogicType } from './draftsLogicType'
+import { loaders } from 'kea-loaders'
+import { HogQLQuery } from '~/queries/schema/schema-general'
+import { lemonToast } from '@posthog/lemon-ui'
+import posthog from 'posthog-js'
+
+export const draftsLogic = kea<draftsLogicType>([
+    path(['scenes', 'data-warehouse', 'editor', 'draftsLogic']),
+
+    actions({
+        saveAsDraft: (
+            query: HogQLQuery,
+            viewId: string,
+            successCallback?: (draft: DataWarehouseSavedQueryDraft) => void,
```

Comment:
In the world of kea, we likely shouldn't be passing around callbacks; much better to use either a `loader()`, or implement a `saveAsDraftSuccess` action